### PR TITLE
Adds shared validators

### PR DESCRIPTION
**Pull request by kylessnell**

### CODE REVIEW

File: dist/src/directives/index.d.ts [hunk 1]
```diff
@@ -1 +1 @@
-export * from './phone_input_formatter.directive';
+export * from './phone_input_formatter.directive';
```
Comment (maulin) [inline dist/src/directives/index.d.ts]: whats the `.d` in the file names?

Comment (kylessnell) [reply dist/src/directives/index.d.ts]: the angular AOT compiler makes these automatically. TS convention i believe.

File: dist/src/directives/phone_input_formatter.directive.d.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
-import { EventEmitter } from '@angular/core';
-export declare class PhoneInputFormatter {
-    phoneNumberChange: EventEmitter<string>;
-    format(event: KeyboardEvent): void;
-}
+import { EventEmitter } from '@angular/core';
+export declare class PhoneInputFormatter {
+    phoneNumberChange: EventEmitter<string>;
```
Comment (maulin) [inline dist/src/directives/phone_input_formatter.directive.d.ts]: extra indentation

File: dist/src/directives/phone_input_formatter.directive.d.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
-import { EventEmitter } from '@angular/core';
-export declare class PhoneInputFormatter {
-    phoneNumberChange: EventEmitter<string>;
-    format(event: KeyboardEvent): void;
-}
+import { EventEmitter } from '@angular/core';
+export declare class PhoneInputFormatter {
+    phoneNumberChange: EventEmitter<string>;
+    format(event: KeyboardEvent): void;
```
Comment (maulin) [inline dist/src/directives/phone_input_formatter.directive.d.ts]: no body for function? Is this an interface?

File: dist/src/pipes/split_phone.pipe.d.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { PipeTransform } from '@angular/core';
-export declare class SplitPhonePipe implements PipeTransform {
-    transform(phoneNumber: string): string;
-}
+import { PipeTransform } from '@angular/core';
+export declare class SplitPhonePipe implements PipeTransform {
+    transform(phoneNumber: string): string;
```
Comment (maulin) [inline dist/src/pipes/split_phone.pipe.d.ts]: indent

File: dist/src/directives/phone_input_formatter.directive.d.ts (hunk 2 above)
Comment (kylessnell) [reply dist/src/directives/phone_input_formatter.directive.d.ts]: This is made automatically by the AOT compiler. :/

**Pull request closed without merging**